enterprise/cypher/slotted-runtime/src/test/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/slotted/pipes/Top1WithTiesSlottedPipeTest.scala

```diff
@@ -0,0 +1,123 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compatibility.v3_3.runtime.slotted.pipes
+
+import org.neo4j.cypher.internal.compatibility.v3_3.runtime.pipes.QueryStateHelper
+import org.neo4j.cypher.internal.frontend.v3_3.symbols._
+import org.neo4j.cypher.internal.frontend.v3_3.test_helpers.CypherFunSuite
+import org.neo4j.cypher.internal.compatibility.v3_3.runtime.slotted.pipes.SlottedValueComparisonHelper._
+
+class Top1WithTiesSlottedPipeTest extends CypherFunSuite {
+
+  test("empty input gives empty output") {
+    val source = new FakeSlottedPipeFromVariables(List(), "x" -> CTAny)
+    val slot = source.pipeline.getReferenceOffsetFor("x")
+    val sortPipe = Top1WithTiesSlottedPipe(source, List(Ascending(slot)))()
+
+    sortPipe.createResults(QueryStateHelper.emptyWithValueSerialization) should be(empty)
+  }
+
+  test("simple sorting works as expected") {
+    val list = List(Map("x" -> "B"), Map("x" -> "A")).iterator
+    val source = new FakeSlottedPipeFromVariables(list, "x" -> CTString)
+    val slot = source.pipeline.getReferenceOffsetFor("x")
+    val sortPipe = Top1WithTiesSlottedPipe(source, List(Ascending(slot)))()
+
+    sortPipe.createResults(QueryStateHelper.emptyWithValueSerialization).toList should beEquivalentTo(List(Map("x" -> "A")))
+  }
+
+  test("three ties for the first place are all returned") {
+    val input = List(
+      Map("x" -> 1, "y" -> 1),
+      Map("x" -> 1, "y" -> 2),
+      Map("x" -> 2, "y" -> 3),
+      Map("x" -> 2, "y" -> 4)
+    ).iterator
+
+    val source = new FakeSlottedPipeFromVariables(input, "x" -> CTInteger, "y" -> CTInteger)
+    val slot = source.pipeline.getReferenceOffsetFor("x")
+    val sortPipe = Top1WithTiesSlottedPipe(source, List(Ascending(slot)))()
+
+    sortPipe.createResults(QueryStateHelper.emptyWithValueSerialization).toList should beEquivalentTo(List(
+      Map("x" -> 1, "y" -> 1),
+      Map("x" -> 1, "y" -> 2)))
+  }
+
+  test("if only null is present, it should be returned") {
+    val input = List(
+      Map[String,Any]("x" -> null, "y" -> 1),
+      Map[String,Any]("x" -> null, "y" -> 2)
+    ).iterator
+
+    val source = new FakeSlottedPipeFromVariables(input, "x" -> CTInteger, "y" -> CTInteger)
+    val slot = source.pipeline.getReferenceOffsetFor("x")
+    val sortPipe = Top1WithTiesSlottedPipe(source, List(Ascending(slot)))()
+
+    sortPipe.createResults(QueryStateHelper.emptyWithValueSerialization).toList should beEquivalentTo(List(
+      Map("x" -> null, "y" -> 1),
+      Map("x" -> null, "y" -> 2)))
+  }
+
+  test("null should not be returned if other values are present") {
```

Comment:
why is `null` treated as some special case?
isn't this entirely up to what the CIP specifies: https://github.com/opencypher/openCypher/blob/master/cip/1.accepted/CIP2016-06-14-Define-comparability-and-equality-as-well-as-orderability-and-equivalence.adoc#orderability-equivalence

Reply:
This test precedes both the new value model and that CIP